PCF-ADR-0002-release-story-and-branch-maintenance.md

```diff
@@ -0,0 +1,138 @@
+- A branching strategy which allows for CI/CD .
+- A way to add bugfixes to existing releases.
+
+## How
+
+### Decision
+
+We have decided to use GitHub Flow as our branching strategy. GitHub Flow is a simple and lightweight model that follows
+these principles:
+
+- The `main` branch is always deployable and represents the latest version of the code.
+- New features and bug fixes are developed in short-lived branches, created from the `main` branch.
+- When a branch is ready for review, a pull request is opened to merge it into the `main` branch.
+- After the pull request is approved by at least one reviewer, it is merged into the `main` branch and deployed to
+  production.
+- If a bugfix is required for any release, the bugfix is added using the above process and then cherry-picked into the
+  branch for the minor release. (explained below)
+
+### Rationale
+
+- It is simple and easy to understand and follow. There is no need to maintain multiple long-lived branches or deal with
+  complex merge conflicts.
+- It supports continuous integration and delivery. Every commit on the `main` branch is tested and deployed
+  automatically, ensuring high quality and fast feedback.
+- It encourages collaboration and code review. Pull requests are the central place for discussing and reviewing code
+  changes, ensuring that every change is approved by at least one other developer before merging.
+- It allows us to release frequently and incrementally. We can deploy new features and bug fixes as soon as they are
+  ready, without waiting for a predefined release cycle or schedule.
+
+### How to pull in bug fixes into a release branch
+
+Sometimes, we may need to create a release branch from the `main` branch to prepare for a major or minor release. For
+example, we may want to freeze the feature set for a certain version and only allow bug fixes on that branch. In that
+case, we can use the following steps to pull in bug fixes from the `main` branch into the release branch:
+
+- Create a release branch from the `main` branch with a name that indicates the minor version number, such as `v1.2`.
+- Continue developing new features and bug fixes on the `main` branch as usual.
+- If a bug fix on the `main` branch needs to be included in the release branch, cherry-pick the commit that contains the
+  fix onto the release branch.
+- When the release branch is ready for deployment, create a tag on GitHub with the patch version incremented the, such
+  as `v1.2.1`. This will trigger the deployment pipeline for that tag.
+
+### Release process
+
+The release process for our project is very simple and consists of creating a version tag on GitHub. For example, if we
+want to release version 1.3 of our software, we just need to create a tag named `v1.3` on GitHub from the latest commit
+on the `main` branch. This will trigger the build and deployment pipelines, or someone will manually take the required
```

Comment:
I'm a little bit confused on the use of release branches and tags. This line states that we would create release tags off of main, but that seems to conflict with the existence of release branches. Would we not want to tag off of release branches? If yes, would we always tag at the point of a release? Can a given release branch have multiple tags (i.e. if we need to do multiple builds/releases for a given version

Reply:
Release branches would only exist to pull in bug fixes to already released versions. At the time of release you would just create a tag like `v0.1.0`, then if there are some bug fixes that we want to patch onto the minor version we create a branch `release-v0.1` from the `v0.1.0` tag. Then cherry pick commits that we want to include in `v0.1.1`. Once all the commits have been pulled in then we can create a release tag `v0.1.1` off of the `release-v0.1` branch. The idea is that feature and bug dev can happen at the same time without having to pull new features into a patch release.

Reply:
This makes a lot of sense. Thanks for the clarification!